ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc

```diff
@@ -0,0 +1,684 @@
+ * @return 0 if OK. non-zero if error.
+ */
+int
+ArmNNCore::getInputTensorDim (GstTensorsInfo * info)
+{
+  gst_tensors_info_copy (info, &inputTensorMeta);
+  return 0;
+}
+
+/**
+ * @brief	return the Dimension of Tensor.
+ * @param[out] info Structure for tensor info.
+ * @todo return whole array rather than index 0
+ * @return 0 if OK. non-zero if error.
+ */
+int
+ArmNNCore::getOutputTensorDim (GstTensorsInfo * info)
+{
+  gst_tensors_info_copy (info, &outputTensorMeta);
+  return 0;
+}
+
+/**
+ * @brief	run the model with the input.
+ * @param[in] input : The array of input tensors
+ * @param[out]  output : The array of output tensors
+ * @return 0 if OK. non-zero if error.
+ */
+int
+ArmNNCore::invoke (const GstTensorMemory * input, GstTensorMemory * output)
+{
+  armnn::InputTensors input_tensors;
+  armnn::OutputTensors output_tensors;
+  armnn::Status ret;
+
+  for (unsigned int i = 0; i < inputTensorMeta.num_tensors; i++) {
+    if (inputBindingInfo[i].second.GetNumBytes () != input[i].size) {
+      input_tensors.clear();
+      return -EINVAL;
+    }
+    armnn::ConstTensor input_tensor (inputBindingInfo[i].second, input[i].data);
+    input_tensors.push_back ( {
+        inputBindingInfo[i].first, input_tensor});
+  }
+
+  for (unsigned int i = 0; i < outputTensorMeta.num_tensors; i++) {
+    if (outputBindingInfo[i].second.GetNumBytes () != output[i].size) {
+      output_tensors.clear();
+      input_tensors.clear();
+      return -EINVAL;
+    }
+    armnn::Tensor output_tensor (outputBindingInfo[i].second, output[i].data);
+    output_tensors.push_back ( {
+        outputBindingInfo[i].first, output_tensor});
+  }
+
+  /** Run the inference */
+  ret = runtime->EnqueueWorkload (networkIdentifier, input_tensors,
+      output_tensors);
+
+  /** Clear the Input and Output tensors */
+  input_tensors.clear();
+  output_tensors.clear();
+
+  if (ret == armnn::Status::Failure)
+    return -EINVAL;
+
+  return 0;
+}
+
+/**
+ * @brief Free privateData and move on.
+ */
+static void
+armnn_close (const GstTensorFilterProperties * prop, void **private_data)
+{
+  ArmNNCore *core;
+
+  core = static_cast < ArmNNCore * >(*private_data);
+  if (core == NULL)
+    return;
+
+  delete core;
+  *private_data = NULL;
+}
+
+/**
+ * @brief The open callback for GstTensorFilterFramework. Called before anything else
+ * @param prop property of tensor_filter instance
+ * @param private_data : armnn plugin's private data
+ */
+static int
+armnn_open (const GstTensorFilterProperties * prop, void **private_data)
+{
+  ArmNNCore *core;
+  accl_hw hw = ACCL_DEFAULT;
+
+  core = static_cast < ArmNNCore * >(*private_data);
+
+  if (core != NULL) {
+    if (g_strcmp0 (prop->model_files[0], core->getModelPath ()) != 0) {
+      armnn_close (prop, private_data);
+    } else {
+      return 1;
+    }
+  }
+
+  if (prop->model_files[0] == NULL)
+    return -EINVAL;
+
+  core = new ArmNNCore (prop->model_files[0], hw);
+  if (core == NULL) {
```

Comment:
`new` does not return NULL. it raises bad_alloc exception

Reply:
Updated. Also updated error handling in loadModel correspondingly.